phoenix-tools/src/main/java/org/apache/phoenix/schema/SchemaExtractionTool.java

```diff
@@ -0,0 +1,98 @@
+package org.apache.phoenix.schema;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.commons.cli.PosixParser;
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.conf.Configured;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+
+import org.apache.hadoop.util.Tool;
+import org.apache.hadoop.util.ToolRunner;
+import java.util.logging.Logger;
+
+public class SchemaExtractionTool extends Configured implements Tool {
+
+    private static final Logger LOGGER = Logger.getLogger(SchemaExtractionTool.class.getName());
+    private static final Option HELP_OPTION = new Option("h", "help",
+            false, "Help");
+    private static final Option TABLE_OPTION = new Option("tb", "table", true,
+            "[Required] Table name ex. table1");
+    private static final Option SCHEMA_OPTION = new Option("s", "schema", true,
+            "[Optional] Schema name ex. schema");
+
+    private String pTableName;
+    private String pSchemaName;
+
+    public static Configuration conf;
+    public String output;
```

Comment:
output should have a getter since you're accessing it lots of times from tests and I assume it's read-only externally.